analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/OldVerifier.cs

```diff
@@ -301,68 +302,104 @@ public static void VerifyNoExceptionThrown(string path, DiagnosticAnalyzer[] dia
                 .AddPaths(RemoveTestCasesPrefix(paths))
                 .Verify();
 
+        [Obsolete("Use VerifierBuilder instead.")]
         public static void VerifyAnalyzer(string path,
                                           DiagnosticAnalyzer[] diagnosticAnalyzers,
-                                          IEnumerable<ParseOptions> options = null,
+                                          ImmutableArray<ParseOptions> options = default,
                                           CompilationErrorBehavior checkMode = CompilationErrorBehavior.Default,
                                           OutputKind outputKind = OutputKind.DynamicallyLinkedLibrary,
                                           IEnumerable<MetadataReference> additionalReferences = null,
                                           DiagnosticDescriptor[] onlyDiagnostics = null) =>
-            VerifyAnalyzer(new[] { path }, diagnosticAnalyzers, options, checkMode, outputKind, additionalReferences, null, onlyDiagnostics);
+            diagnosticAnalyzers.Aggregate(new VerifierBuilder(), (builder, analyzer) => builder.AddAnalyzer(() => analyzer))
```

Comment:
That's a lot of builders created without a real need. I think we should either change `AddAnalyzer` to be `AddAnalyzers` and accept more analyzers, or add an overload.

Reply:
This `Aggregate` hack should only exist in the `OldVerifier` so I would not add a special overload for it. And I plan to remove these in the 1st round of cleanup - there's a bunch of overloads used only a few times that I'd like to remove in another PR.
This specific one is used only once with 2 analyzers. So it's not very expensive.